XenAdmin/Wizards/NewNetworkWizard.cs

```diff
@@ -197,6 +205,25 @@ private void CreateCHIN()
             (new CreateChinAction(xenConnection, network, theInterface)).RunAsync();
         }
 
+        private void CreateSRIOV()
+        {
+            XenAPI.Network network = PopulateNewNetworkObj();
+
+            List<PIF> sriovSelectedPifs = new List<PIF>();
+            Pool pool = Helpers.GetPoolOfOne(Host.Connection);
+            foreach (Host host in pool.Connection.Cache.Hosts)
+            {
+                foreach (PIF thePIF in Host.Connection.ResolveAll<PIF>(host.PIFs))
+                {
+                    if (thePIF.IsPhysical() && !thePIF.IsBondNIC() && thePIF.SriovCapable() && thePIF.Name() == pageSriovDetails.SelectedHostNic.Name() && thePIF.sriov_physical_PIF_of.Count == 0)

```

Comment:
In the NetWSriovDetails page lists the NICs' name, those listed NICs' are physical and support SR-IOV. We match the name here because we should enable SR-IOV on those NICs that user choose in the NetWSriovDetails page. The truth is I didn't find other field can do this.

Reply:
How about pif.device?